From 8e9d28c098dff083691e7d5a1f7d42f4fcd697f8 Mon Sep 17 00:00:00 2001
From: Robert Schulze <robert@clickhouse.com>
Date: Wed, 19 Jun 2024 08:40:16 +0000
Subject: [PATCH 021/125] Incorporate more review feedback

Reviewed-by: Neil Horman <nhorman@openssl.org>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
Reviewed-by: Matt Caswell <matt@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/24673)

(cherry picked from commit af82623d32962b3eff5b0f0b0dedec5eb730b231)
---
 ssl/ssl_local.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ssl/ssl_local.h b/ssl/ssl_local.h
index 140e8cf260..7bdd6fbddc 100644
--- a/ssl/ssl_local.h
+++ b/ssl/ssl_local.h
@@ -532,7 +532,6 @@ struct ssl_session_st {
      * certificate is not ok, we must remember the error for session reuse:
      */
     long verify_result;         /* only for servers */
-    CRYPTO_REF_COUNT references;
     OSSL_TIME timeout;
     OSSL_TIME time;
     OSSL_TIME calc_timeout;
@@ -574,9 +573,10 @@ struct ssl_session_st {
 
     /*
      * These are used to make removal of session-ids more efficient and to
-     * implement a maximum cache size.
+     * implement a maximum cache size. Access requires protection of ctx->lock.
      */
     struct ssl_session_st *prev, *next;
+    CRYPTO_REF_COUNT references;
 };
 
 /* Extended master secret support */
-- 
2.39.5 (Apple Git-154)

